assets/src/design-system/components/input/index.js

```diff
@@ -145,6 +173,7 @@ Input.propTypes = {
   disabled: PropTypes.bool,
   hasError: PropTypes.bool,
   hint: PropTypes.string,
+  Icon: PropTypes.elementType,
```

Comment:
Instead of passing an icon component definition, we could pass the component itself. This could even be wrapped in a button for click-ability:
```suggestion
  Icon: PropTypes.element,
```